Model/Ui/OmiseConfigProvider.php

```diff
@@ -5,8 +5,10 @@
 use Magento\Payment\Model\CcConfig as MagentoCcConfig;
 use Omise\Payment\Model\Config\Cc as OmiseCcConfig;
 use Omise\Payment\Model\Customer;
+use Omise\Payment\Model\Capabilities;
```

Comment:
Whether or not the option to switch instalment payments on/off is available in admin should surely be dependent upon the existence of instalment backends within the account's `capabilities` object?

Reply:
Sound like we are speaking the same thing..
(the current code is fetching Capabilities API every time when Magento displays the checkout page. Instead we (maybe) can fetch this Installment backends when the Installment payment method is enabled only)

Reply:
Similar, but they should only be able to turn it on in admin IF their account has some instalment backends, or I guess we can leave the switch there, but display a message if their account has no instalment backends. If we don't do this, we're going to get spurious "instalment payments aren't working" support requests. A simple information message will avoid these

Reply:
Got your point now, yeah, would be great if we can do that 
(it would also solve those legacy issues where "merchant enable 3-D Secure, IB, Alipay only from Magento side but not from Omise side" as well). _(maybe in the future PR?)_

Anyway, that is different from the "fetching Capabilities data every time Magento displays the check out page" topic.
This class will be executed only in the frontend side. And this code, `'capabilities' => $this->capabilities->get()` still need to be here to pass the instalment backends to JS file. Just need to make sure that it won't fetch the data (as it's unnecessary) if the payment method is disabled.

Reply:
If we don't call it every time, and don't have the API response stored, how will we know which instalment methods to display?

Reply:
Agreed